### ISSUE TITLE
Include bpmnProcessId as part of exported variable records

### ISSUE THREAD

**Issue by RomanJRW** [camunda-cloud/zeebe#8181]
**Is your feature request related to a problem? Please describe.**
When importing variable records and converting them to Optimize variables, we have to fetch the definition from ES to extract the `bpmnProcessId` in order to route it to the correct Optimize index. This can be expensive, and results in retries if the process hasn't been been exported by zeebe/imported by Optimize. Furthermore, if the process has been exported but the record is no longer available (cleaned up), then the import gets stuck, as seen in a recent support issue.

**Describe the solution you'd like**
If the `bpmnProcessId` (like how it is included on the incident records for example) was included as part of the exported variable record, we could improve the performance of import by removing the fetch, and also guard against the possibility of the import getting stuck. If we have the key, Optimize is resilient enough to continue the import even if that process never can be imported.

**Describe alternatives you've considered**
The current implementation sort of is our alternative, but it is not ideal. The performance hit is somewhat acceptable, but the import blocking issue is more awkward for us to work around. With some refactoring, skipping this data import after x retries could potentially avoid the skip but also exposes the risk of missing data that the user would expect in Optimize

**Additional context**
Raised as a result of investigation from https://jira.camunda.com/browse/SUPPORT-12111

**Comment by RomanJRW** [comment]
Exactly. We store all instance data for a given definition across _all_ versions (so by the `bpmnProcessId`) to a single index. We still store the key and other properties to the document, but it is this `bpmnProcessId` that is used to create the index or route documents to the correct index.

One of the reasons for this is that in Optimize, it is quite typical for a report to evaluate instances for a given process from across all versions of that process, so it makes sense for these to exist in one index. Another is that splitting this up by version would/could result in too many indices. We have runtime engine customers that have thousands of definitions, with even hundreds of versions for each definition.

**Comment by menski** [comment]
Thanks Josh, we try to solve this during the quarter, but it doesn't have the highest priority. Please let us know if this is an blocking  issue so that we have to increase the priority.

**Comment by RomanJRW** [comment]
Thanks @menski. So far we only know of it affecting one customer - if it becomes more frequent we will let you know that it needs bumping 👍

**Comment by RomanJRW** [comment]
Hi @menski, do you think this is likely to be picked up in Q1? We have a focus on import performance this quarter and resolving this would be a nice improvement for Optimize. Thanks 🙂

**Comment by npepinpe** [comment]
I can make sure it's there for alpha2 at the latest. Of course we're also happy to review a PR from someone on your team :wink:
